### PR TITLE
add capability to fetch a wallet

### PR DESCRIPTION
**Pull request by MrHamdulay**

### CODE REVIEW

**Comment by MrHamdulay** [comment]
@smn @hodgestar

**Comment by MrHamdulay** [comment]
incorrect base :/ but it gives the best looking diff until issue-1 is merged

File: src/app.js [hunk 1]
```diff
@@ -10,7 +10,11 @@ server.use(Restify.bodyParser({ mapParams: true }));
 
 
 server.post('/v1/wallet', WalletController.create);
+server.get('/v1/wallet/:msisdn', WalletController.fetch);
 
 server.listen(Config.PORT, function() {
 	console.log('Listening for requests');
 });
+server.on('uncaughtException', function (req, res, route, err) {
+    console.log('uncaughtException', err.stack);
```
Comment (smn) [inline src/app.js]: `console.error`?

File: src/models/wallet.js [hunk 2]
```diff
@@ -5,7 +5,33 @@ var Sjcl = require('sjcl');
...
-var wallet = {
+var Wallet = {
+    fetch: function(msisdn, pin) {
+        // fetch salt, calculate pin hash, validate pin hash
+        return DbUtil.promiseConnection()
+            .then(DbUtil.chainedQuery(
+                "SELECT address, publickey, privatekey, pinhash, salt \
+                 FROM wallet \
+                 WHERE msisdn = $1", [msisdn]))
+            .then(function(result) {
```
Comment (smn) [inline src/models/wallet.js]: What's this? Do we want to use sequelize here?

File: src/models/wallet.js [hunk 3]
```diff
@@ -19,8 +45,10 @@ var wallet = {
 
         return DbUtil.promiseConnection()
             .then(DbUtil.chainedQuery(
-                "INSERT INTO wallet ( msisdn, address, publickey, privatekey, pinhash, salt) VALUES ( $1, $2, $3, $4, $5, $6)",
-                [msisdn, pinHash, address, publicKey, privateKeyEncrypted, salt]))
+                "INSERT INTO wallet  \
+                    (msisdn, address, publickey, privatekey, pinhash, salt) \
+                VALUES ( $1, $2, $3, $4, $5, $6)",
+                [msisdn, address, publicKey, privateKeyEncrypted, pinHash, salt]))
```
Comment (smn) [inline src/models/wallet.js]: Ditto.

File: src/utils/db.js [hunk 4]
```diff
@@ -31,9 +35,17 @@ var db = {
...
+                            error: error,
+                            connection: connection
+                        });
                     } else {
-                        resolve(result);
+                        resolve({
+                            success: true,
+                            result: result,
+                            connection: connection
+                        });
```
Comment (smn) [inline src/utils/db.js]: Don't think we need this anymore with sequelize.

**Pull request closed without merging**